lib/potlock/client.rb

```diff
@@ -29,14 +30,19 @@ def get
 
     def set(&block)
       value = lock!(&block)
-      redis.set(key, value)
+      store_value!(value)
```

Comment:
Isn't it weird that the `redis.set` call is not inside the `lock!` block? This feels like a bug of the gem 🤔 It looks like it should be
```suggestion
      lock! do
        value = block.call
        store_value!(value)
      end
```

Otherwise, I think it would be _technically_ possible that a process that started later than another ends up storing its value in Redis first, so it would be overwritten by the first process.

Reply:
Agree, good catch